.github/workflows/codeql.yml

```diff
@@ -0,0 +1,74 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "sycl-develop" ]
+  pull_request:
+    branches: [ "sycl-develop" ]
+  schedule:
+    - cron: '23 5 * * 5'
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    permissions:
+      # required for all workflows
+      security-events: write
+      # required to fetch internal or private CodeQL packs
+      packages: read
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - language: c-cpp
+          build-mode: manual
+        - language: python
+          build-mode: none
+
+    container:
+      image: nvidia/cuda:12.4.1-devel-ubuntu22.04
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
+      with:
+        languages: ${{ matrix.language }}
+        build-mode: ${{ matrix.build-mode }}
+
+    - name: Install dependencies
+      if: matrix.build-mode == 'none'
+      shell: bash
+      run: |
+        apt update && apt install -y python3
+
+    - name: Build
+      if: matrix.build-mode == 'manual'
+      shell: bash
+      run: |
+        apt update && apt install -y cmake git libstdc++-12-dev ninja-build python3 wget && \
+        pushd /usr/local && \
+        wget -q https://github.com/intel/llvm/releases/download/nightly-2024-05-22/sycl_linux.tar.gz && \
```

Comment:
We'll likely need to update the nightly version frequently. Would it be possible to define the version in a global variable so that it can be reused for all workflows?